tests/solver_tester.py

```diff
@@ -11,227 +11,79 @@
-                return system.run(tsave, solver, options=options, H=H, L=L, y0=y0)
-
-            # batched L
-            result = run(Hb=False, Lb=True, yb=False)
-            assert result.ysave.shape == (bL, nt, n, n)
-            assert result.Esave.shape == (bL, nE, nt)
-
-            # batched H and L
-            result = run(Hb=True, Lb=True, yb=False)
-            assert result.ysave.shape == (bH, bL, nt, n, n)
-            assert result.Esave.shape == (bH, bL, nE, nt)
-
-            # batched L and y0
-            result = run(Hb=False, Lb=True, yb=True)
-            assert result.ysave.shape == (bL, by, nt, n, n)
-            assert result.Esave.shape == (bL, by, nE, nt)
-
-            # batched H and L and y0
-            result = run(Hb=True, Lb=True, yb=True)
-            assert result.ysave.shape == (bH, bL, by, nt, n, n)
-            assert result.Esave.shape == (bH, bL, by, nE, nt)
-
-        # === test non cartesian batching
-        options = {} if options is None else options
-        options['cartesian_batching'] = False
-        b = 2
-
-        Hb = system.Hb[:b]
-        y0b = system.y0b[:b]
-
-        if isinstance(system, ClosedSystem):
-            result = system.run(tsave, solver, options=options, H=Hb, y0=y0b)
-            assert result.ysave.shape == (b, nt, n, 1)
-            assert result.Esave.shape == (b, nE, nt)
-        elif isinstance(system, OpenSystem):
-            Lb = [L[:b] for L in system.Lb]
-            result = system.run(tsave, solver, options=options, H=Hb, L=Lb, y0=y0b)
-            assert result.ysave.shape == (b, nt, n, n)
-            assert result.Esave.shape == (b, nE, nt)
-
     def _test_correctness(
         self,
         system: System,
         solver: Solver,
         *,
         options: dict[str, Any] | None = None,
-        ntsave: int = 11,
         ysave_atol: float = 1e-3,
         esave_rtol: float = 1e-3,
         esave_atol: float = 1e-5,
     ):
-        tsave = system.tsave(ntsave)
-        result = system.run(tsave, solver, options=options)
+        result = system.run(solver, options=options)
 
         # === test ysave
-        errs = jnp.linalg.norm(result.ysave - system.states(tsave), axis=(-2, -1))
+        true_ysave = system.states(system.tsave)
+        errs = jnp.linalg.norm(true_ysave - result.ysave, axis=(-2, -1))
         logging.warning(f'errs = {errs}')
         assert jnp.all(errs <= ysave_atol)
 
         # === test Esave
-        true_Esave = system.expects(tsave)
-        logging.warning(f'Esave      = {result.Esave}')
+        true_Esave = system.expects(system.tsave)
         logging.warning(f'true_Esave = {true_Esave}')
-        assert jnp.allclose(result.Esave, true_Esave, rtol=esave_rtol, atol=esave_atol)
+        logging.warning(f'Esave      = {result.Esave}')
+        assert jnp.allclose(true_Esave, result.Esave, rtol=esave_rtol, atol=esave_atol)
 
     def test_correctness(self):
         pass
 
-    def _test_jit(
```

Comment:
Why remove the jit test?

Reply:
I don't see the point of this test? If it's to test that the code works with JIT, it's jitted by default so already tested by the rest. If it's to test wether the jitted code runs faster, the answer will always be yes (I don't see a bug/situation where it's not the case). Moreover the no-jit code takes quite long to run (on the order of a few seconds), so once we test all the solvers it would slow down the CI quite a lot.